File: practice/Leetcode/3.无重复字符的最长子串/code.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        lenth = len(s)
        tmp = ''
        result = 0
        for i in range(lenth):
            flag = tmp.find(s[i])
            tmp += s[i]
            if flag != -1:
                result = max(len(tmp) - 1, result)
                tmp = tmp[flag + 1:]
        result = max(len(tmp), result)
        return result

File: practice/Leetcode/3.无重复字符的最长子串/code.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        char occupied[256] = {0};
        int max_len = 0;
        int len = s.length();
        int left = 0, right = 0;
        while(left < len && right < len){
            while (occupied[s[right]]){
                max_len = max_len > (right - left) ? max_len : (right - left);
                occupied[s[left]] = 0;
                left ++;
            }
            occupied[s[right]] = 1;
            right++;
        }
        return max_len > (right - left) ? max_len : (right - left);
    }
};